packages/ember-data/lib/system/store.js

```diff
@@ -1847,15 +1862,12 @@ Store = Service.extend({
 
     @method adapterFor
     @private
-    @param {String or subclass of DS.Model} type
+    @param {String} modelName
     @return DS.Adapter
   */
-  adapterFor: function(type) {
-    if (type !== 'application') {
-      type = this.modelFor(type);
-    }
-
-    var adapter = this.lookupAdapter(type.modelName) || this.lookupAdapter('application');
+  adapterFor: function(modelName) {
+    Ember.assert('Passing classes to store methods has been removed. Please pass a dasherized string instead of '+ Ember.inspect(modelName), typeof modelName === 'string');
```

Comment:
I fixed the places our own code was doing it.. Seems odd to ship a version that triggers deprecation warnings?